Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAccountProtectionLocalAdministratorPasswordSolutionPolicy/MSFT_IntuneAccountProtectionLocalAdministratorPasswordSolutionPolicy.schema.mof

```diff
@@ -0,0 +1,37 @@
+ [Write, Description("The Id of the filter for the target assignment.")] String deviceAndAppManagementAssignmentFilterId;
+ [Write, Description("The group Id that is the target of the assignment.")] String groupId;
+ [Write, Description("The collection Id that is the target of the assignment.(ConfigMgr)")] String collectionId;
+};
+
+[ClassVersion("1.0.0.0"), FriendlyName("IntuneAccountProtectionLocalAdministratorPasswordSolutionPolicy")]
+class MSFT_IntuneAccountProtectionLocalAdministratorPasswordSolutionPolicy : OMI_BaseResource
+{
+ [Key, Description("Identity of the account protection local administrator password solution policy.")] String Identity;
+ [Required, Description("Display name of the account protection local administrator password solution policy.")] String DisplayName;
+ [Write, Description("Description of the account protection local administrator password solution policy.")] String Description;
+ [Write, Description("Assignments of the account protection local administrator password solution policy."), EmbeddedInstance("MSFT_DeviceManagementConfigurationPolicyAssignments")] String Assignments[];
+ [Write, Description("Configures which directory the local admin account password is backed up to. 0 - Disabled, 1 - Azure AD, 2 - AD"), ValueMap{"0", "1", "2"}, Values{"0", "1", "2"}] UInt32 BackupDirectory;
+ [Write, Description("Configures the maximum password age of the managed local administrator account for Azure AD. Minimum - 7, Maximum - 365")] UInt32 PasswordAgeDays_AAD;
+ [Write, Description("Configures the maximum password age of the managed local administrator account for Active Directory. Minimum - 1, Maximum - 365")] UInt32 PasswordAgeDays;
+ [Write, Description("Configures additional enforcement of maximum password age for the managed local administrator account.")] Boolean PasswordExpirationProtectionEnabled;
+ [Write, Description("Configures how many previous encrypted passwords will be remembered in Active Directory. Minimum - 0, Maximum - 12")] UInt32 AdEncryptedPasswordHistorySize;
+ [Write, Description("Configures whether the password is encrypted before being stored in Active Directory.")] Boolean AdPasswordEncryptionEnabled;
+ [Write, Description("Configures the name or SID of a user or group that can decrypt the password stored in Active Directory.")] String AdPasswordEncryptionPrincipal;
+ [Write, Description("Configures the name of the managed local administrator account.")] String AdministratorAccountName;
+ [Write, Description("Configures the password complexity of the managed local administrator account. 1 - Large letters, 2 - Large + small letters, 3 - Large + small letters + numbers, 4 - Large + small letters + numbers + special characters"), ValueMap{"1", "2", "3", "4"}, Values{"1", "2", "3", "4"}] UInt32 PasswordComplexity;
```

Comment:
From what I've read and understood the `Values` keyword is for limiting the input elements to a subset of elements, and the `ValueMap` for mapping specific indexes to the set of values. The only way I have found to make the possible elements appear in the suggestion in VS Code is by filling the possible values into the `ValueMap` keyword. 
Although the definition in the `M365TenantConfig.ps1` script will include the integers as strings, the compiled MOF file will show them as integers. 

Some other modules like `MSFT_IntuneAntivirusPolicyWindows10SetingsCatalog` or `MSFT_IntuneSettingCatalogASRRulesPolicyWindows10` utilize the same approach for defining specific values applicable for a property, albeit for a String value. 

I didn't find any specific Microsoft or dmtf specification about how to properly handle an integer restriction for a property, so I'm wondering if there is any or if the validation should be solely done in the PowerShell code. But wouldn't that be misleading in a way that the MOF successfully compiles but then cannot be executed by the DSC configuration? Just my opinion, but I would rather configure an integer value as a "string" with autocompletion and MOF validation rather than having a field that accepts any integer and fails when executing the DSC configuration. 

Additionally, most of these restricted values for a property are extracted from a longer string, where the integer value is the last letter in that string. But because it was extracted from a string and not parsed to an integer value, it still is a string and also gets exported as one. That does not change, even without the `Values` and `ValueMap` keywords. The MOF compilation still succeeds though, since `"1"` gets converted to `1`.

Reply:
@ykuijs Any update? Instead of `UInt32` I could also set the data type to `string`, if that simplifies the validation.

Reply:
You are absolutely right. Was just wondering if that would work. Just check some other code and it is done exactly the same there as well 👍